src/platform/utilities/api/index.js

```diff
@@ -3,6 +3,33 @@ import * as Sentry from '@sentry/browser';
 import environment from '../environment';
 import localStorage from '../storage/localStorage';
 
+export function fetch(...args) {
+  // Only replace custom fetch if fetch is not stubbed and
+  // feature flag is enabled
+  if (
+    global.fetch.displayName !== 'stub' &&
+    localStorage.getItem('useCustomFetch') === 'true'
+  ) {
+    return global.fetch.apply(this, args).then(response => {
+      const apiURL = environment.API_URL;
+
+      if (
+        response.url.includes(apiURL) &&
+        (response.ok || response.status === 304)
+      ) {
+        // Get session expiration from header
+        const sessionExpiration = response.headers.get('X-Session-Expiration');
+        if (sessionExpiration) {
+          localStorage.setItem('sessionExpiration', sessionExpiration);
+        }
+      }
+      return response;
+    });
+  }
+
+  return global.fetch(...args);
+}
```

Comment:
I feel a little weird about needing to reference the global fetch. Maybe we can name this function as something like `fetchAndUpdateSessionExpiration`, but then export it on a different line like `export { fetch: fetchAndUpdateSessionExpiration }`. Or, maybe we can export it as `fetchAndUpdateSessionExpiration` and import it in other modules like `import { fetchAndUpdateSessionExpiration as fetch } from '../../utilities/api'`. What do you think?